xdsl/irdl/declarative_assembly_format.py

```diff
@@ -18,7 +20,9 @@
     ParametrizedAttribute,
     SSAValue,
     TypedAttribute,
+    Region,
 )
+from xdsl.ir.core import Block, Operation
```

Comment:
Please install the precommit, this shouldn't pass the linter

Reply:
That's the part I'm not completely sure about from the documentation, I know that the generic format does this, and that the attribute dict has to be attr-dict-with-keyword, but am unsure if its allowed to have the attributes in the middle of regions (but labelled).

Reply:
I tried some stuff locally and the format "$body attr-dict-with-keyword" seems to work absolutely fine

Reply:
Hmm okay, maybe its because I'm running an older version, will fix

Reply:
From my testing it really only seems to complain if you have an attr-dict immediately followed by a region

Reply:
https://github.com/llvm/llvm-project/blob/ea28668f8c980c9987b0e2f305b0a8eff9797401/mlir/tools/mlir-tblgen/OpFormatGen.cpp#L3010 Here is the check

Reply:
As in does not accept "attr-dict $body" but does accept "attr-dict-with-keyword $body"?

Reply:
Yes, it looks like the only thing that is denied is attr-dict followed directly by a region variable

Reply:
I also tested ```attr-dict `,` $body``` and that worked as expected too.

Reply:
Have made these updates and added tests, should work now